EMongoModel.php

```diff
@@ -404,10 +404,15 @@ public function getRelated($name, $refresh = false, $params = array())
 				$result = array();
 				foreach($pk as $singleReference){
 					$row = $this->populateReference($singleReference, $cname);
-					if($row){
-						array_push($result, $row);
-					}
+					
+					// When $row does not exists it will return null. It will not add it to $result

```

Comment:
Are you sure about this? I just ran a quick tester and showed the result of [0 => null] http://sandbox.onlinephpfunctions.com/code/c25fa888f2340e65714e801ed7e2a9077f83fa1d